server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -138,3 +170,149 @@ func getOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Reque
 		return
 	}
 }
+
+func createOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Request) {
+	service, ok := ensureOutgoingOAuthConnectionInterface(c, whereOutgoingOAuthConnection)
+	if !ok {
+		return
+	}
+
+	if !checkOutgoingOAuthConnectionWritePermissions(c) {
+		return
+	}
+
+	var inputConnection model.OutgoingOAuthConnection
+	if err := json.NewDecoder(r.Body).Decode(&inputConnection); err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.create_connection.input_error", nil, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	inputConnection.CreatorId = c.AppContext.Session().UserId
+
+	auditRec := c.MakeAuditRecord("createOutgoingOauthConnection", audit.Fail)
```

Comment:
Should the audit come before permissions checks? We may still want to track a failed request.